kubesphere/ks-devops
# Fix that the controller did not use creator's token to trigger a Jenkins job

### What dose this PR?

1. Add `devops.kubesphere.io/creator` into annotations of PipelineRun
2. Trigger PipelineRun with specific token which is issued by `devops.kubesphere.io/creator`

### Why do we need it?

At present, if we trigger a Pipeline using non-admin account, we will get message `Started by admin` always.

### Steps to test

Docker images for test:

```bash
johnniang/devops-apiserver:dev-v3.2.1-rc.1-2debd81
johnniang/devops-controller:dev-v3.2.1-rc.1-2debd81
```
1. Create a Workspace, DevOps Project, Pipeline using admin account
1. Create another account tester and invite him/her into that Workspace and DevOps Project
1. Trigger that Pipeline using tester account
1. See the `Last Message` column

![image](https://user-images.githubusercontent.com/16865714/143803721-bad38141-8d49-4c32-9328-5bc138b9ac40.png)

/kind bug
/milestone v3.2
/cc @kubesphere/sig-devops
